airflow-core/src/airflow/api_fastapi/app.py

```diff
@@ -49,6 +49,9 @@
 # Define the full path on which the potential auth manager fastapi is mounted
 AUTH_MANAGER_FASTAPI_APP_PREFIX = f"{API_ROOT_PATH}auth"
 
+# Fast API apps mounted under these prefixes are not allowed
+RESERVED_URL_PREFIXES = {"/api/v2", "/ui", "/execution"}
```

Comment:
Since we will only iterate though the element, we can simply define as list instead of set.
```suggestion
RESERVED_URL_PREFIXES = ["/api/v2", "/ui", "/execution"]
```